eureka-client/src/main/java/com/netflix/appinfo/InstanceInfo.java

```diff
@@ -368,21 +377,26 @@ public boolean equals(Object obj) {
 
         private String namespace;
 
-        private Builder(InstanceInfo result, VipAddressResolver vipAddressResolver) {
+        private Builder(InstanceInfo result, VipAddressResolver vipAddressResolver, Function<String,String> intern) {
             this.vipAddressResolver = vipAddressResolver;
             this.result = result;
+            this.intern = intern != null ? intern : StringCache::intern;
```

Comment:
there are three other pre-existing usages (constructor + 2 'newBuilder' factory methods) that call this new private constructor.  Each of those usages would have to specify StringCache::intern as the interning function, instead of having the default established in a single place.